pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark-3/src/main/java/org/apache/pinot/plugin/ingestion/batch/spark3/SparkSegmentMetadataPushJobRunner.java

```diff
@@ -91,43 +131,123 @@ public void run() {
         segmentsToPush.add(file);
       }
     }
+    Map<String, String> segmentsToPushUriToTarPathMap =
```

Comment:
there's a lot of interleaved stuff here: multiple branches and things only used in some branches

It feels like we need 1 branch point
```
if (consistentPushEnabled) {
  pushSegmentsWithConsistentDataPush();
} else {
  pushSegments();
}
```